src/GroupManager.php

```diff
@@ -230,11 +268,85 @@ public function removeGroup($entity_type_id, $bundle_id) {
 
       // Only update and refresh the map if a key was found and unset.
       $editable->set('groups', $groups);
-      $saved = $editable->save();
+      $editable->save();
+
+      // Remove all roles associated with this group type.
+      $this->removeRoles($entity_type_id, $bundle_id);
 
       $this->resetGroupMap();
+    }
+  }
+
+  /**
+   * Creates default roles for the given group type.
+   *
+   * @param string $entity_type_id
+   *   The entity type ID of the group for which to create default roles.
+   * @param string $bundle_id
+   *   The bundle ID of the group for which to create default roles.
+   *
+   * @todo: Would a dedicated RoleManager service be a better place for this?
+   */
+  protected function createRoles($entity_type_id, $bundle_id) {
+    foreach ($this->getDefaultRoles() as $role_name => $default_properties) {
+      $properties = [
+        'group_type' => $entity_type_id,
+        'group_bundle' => $bundle_id,
+        'id' => $role_name,
+        'role_type' => OgRole::getRoleTypeByName($role_name),
+      ];
+
+      // Populate the default permissions.
+      $event = new PermissionEvent($entity_type_id, $bundle_id);
+      /** @var \Drupal\og\Event\PermissionEventInterface $permissions */
+      $permissions = $this->eventDispatcher->dispatch(PermissionEventInterface::EVENT_NAME, $event);
+      $properties['permissions'] = array_keys($permissions->filterByDefaultRole($role_name));
+
+      $role = $this->ogRoleStorage->create($properties + $default_properties);
+      $role->save();
+    }
+  }
+
+  /**
+   * Returns the default roles.
+   *
+   * @return array
+   *   An associative array of default role properties, keyed by role name. Each
+   *   role property is an associative array with the following keys:
+   *   - 'label': The human readable label.
+   *   - 'role_type': Either OgRoleInterface::ROLE_TYPE_STANDARD or
+   *     OgRoleInterface::ROLE_TYPE_REQUIRED.
+   *
+   * @todo: Would a dedicated RoleManager service be a better place for this?
+   */
+  public function getDefaultRoles() {
+    /** @var \Drupal\og\Event\DefaultRoleEvent $default_role_event */
+    $event = new DefaultRoleEvent();
+    $default_role_event = $this->eventDispatcher->dispatch(DefaultRoleEventInterface::EVENT_NAME, $event);
+
+    // @todo: Do we want projects to be able to override the two required
```

Comment:
in OG7 we don't allow changing the label, so I think it's fine also to prevent that in OG8